### PR TITLE
fix(webview): webview API migration

### PR DESCRIPTION
**Pull request by invincibleJai**
- Migrating to VSCode webview API from previewhtml

https://code.visualstudio.com/api/extension-guides/webview

- refactor of auth module to make use of promises
- updates test

### CODE REVIEW

File: src/dependencyReportPanel.ts [hunk 1]
```diff
@@ -0,0 +1,190 @@
+import * as path from 'path';
+import * as vscode from 'vscode';
+
```
Comment (sudhirverma) [inline src/dependencyReportPanel.ts]: unit test for this file?

Comment (invincibleJai) [reply src/dependencyReportPanel.ts]: added, PTAL :)

**Comment by invincibleJai** [comment]
[test]

File: src/dependencyReportPanel.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            vscode.window.showErrorMessage(message.text);
+            return;
+        }
+      },
+      null,
+      this._disposables
+    );
+  }
+
+  public doUpdatePanel(data) {
```
Comment (sudhirverma) [inline src/dependencyReportPanel.ts]: can you declare the type of parameter?

File: src/multimanifestmodule.ts [hunk 3]
```diff
@@ -235,28 +212,17 @@ export module multimanifestmodule {
...
   };
 
-  export const triggerFullStackAnalyses = (
-    context,
-    workspaceFolder,
-    provider,
-    previewUri
-  ) => {
-    provider.signalInit(previewUri, null);
+  export const triggerFullStackAnalyses = (context, workspaceFolder) => {
```
Comment (sudhirverma) [inline src/multimanifestmodule.ts]: same here type of parameter.

File: src/stackAnalysisService.ts [hunk 4]
```diff
@@ -104,4 +104,29 @@ export module stackAnalysisServices {
       });
     });
   };
+
+  export const get3ScaleRouteService = function(options) {
```
Comment (sudhirverma) [inline src/stackAnalysisService.ts]: can you provide type of parameter

**Pull request merged**